lib/views/after_auth_screens/events/event_calendar.dart

```diff
@@ -156,6 +154,25 @@ _AppointmentDataSource _getCalendarDataSource(List<Event> eventsList) {
   return _AppointmentDataSource(appointments);
 }
 
+/// a_line_ending_with_end_punctuation.
+///
+/// **params**:
+/// * `time`: define_the_param
+///
+/// **returns**:
+/// * `DateTime`: define_the_return
+DateTime _parseTime(String time) {
+  try {
+    return DateFormat('h:mm a', 'en_US').parse(time);
+  } on FormatException {
+    print('Caught FormatException: $time');
+    return DateFormat('Hms').parse(time);
+  } catch (e) {
+    print('Caught error: $e');
+    throw Exception('Invalid time format: $time');
```

Comment:
Write the tests for these lines.

Reply:
I have added test for these lines please review